### PR TITLE
Balances adjustment

### PR DESCRIPTION
**Pull request by WalAmeni**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Add balances adjustment feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
no breaking change

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by murgeyseb** [comment]
> Could you remove all target/* files that have been committed in this PR?

And add the associated lines in .gitignore file.

File: powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        String initialVariantId = network.getVariantManager().getWorkingVariantId();
+        String workingVariantCopyId = workingStateId + " COPY";
+        network.getVariantManager().cloneVariant(workingStateId, workingVariantCopyId);
+        network.getVariantManager().setWorkingVariant(workingVariantCopyId);
+
+        Map<NetworkArea, Double> networkAreasResidue;
+        List<NetworkArea> unbalancedNetworkAreas;
+
+        while (iterationCounter < parameters.getMaxNumberIterations()
+                && BalanceComputationResult.Status.FAILED.equals(result.getStatus())) {
```
Comment (murgeyseb) [inline powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java]: Prefer result.getStatus() == BalanceComputationResult.Status.FAILED

File: powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                network.getVariantManager().cloneVariant(workingStateId, workingVariantCopyId);
+                network.getVariantManager().setWorkingVariant(workingVariantCopyId);
+                LOGGER.info(" Scaling iteration number {}", iterationCounter);
+
+                Map<NetworkArea, Double> scaleNetworkAreasMap = scaleNetworkAreas(networkAreasResidue, previousScalingMap);
+                previousScalingMap = scaleNetworkAreasMap;
+            }
+        }
+
+        if (BalanceComputationResult.Status.SUCCESS.equals(result.getStatus())) {
```
Comment (murgeyseb) [inline powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java]: Prefer result.getStatus() == BalanceComputationResult.Status.SUCCESS

File: powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+            List<String> networkAreasName = networkAreaNetPositionTargetMap.keySet().stream()
+                    .map(networkArea -> networkArea.getName()).collect(Collectors.toList());
+            LOGGER.info(" Network areas : {} are balanced after {} iterations", networkAreasName, result.getIterationCount());
+
+        } else {
+            // return the network at initial working state id
+            network.getVariantManager().removeVariant(workingVariantCopyId);
+            network.getVariantManager().cloneVariant(workingStateId, workingVariantCopyId);
+            network.getVariantManager().setWorkingVariant(workingVariantCopyId);
+            LOGGER.error(" Network areas are unbalanced after 5 iterations");
```
Comment (murgeyseb) [inline powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java]: replace "5" by "iterationCounter "

File: powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java (hunk 1 above)
Comment (WalAmeni) [reply powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java]: Ok

File: powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java (hunk 2 above)
Comment (WalAmeni) [reply powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java]: Ok

File: powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java (hunk 3 above)
Comment (WalAmeni) [reply powsybl-balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java]: OK

**Pull request merged**